lib/wallet/open_channel.dart

```diff
@@ -25,126 +24,149 @@ class OpenChannel extends StatefulWidget {
+                              style: const TextStyle(color: Colors.grey)),
+                          TextFormField(
+                            initialValue: ThousandsSeparatorInputFormatter()
+                                .formatEditUpdate(TextEditingValue.empty,
+                                    TextEditingValue(text: maxTakerChannelAmount.toString()))
+                                .text,
+                            readOnly: submitting,
+                            keyboardType: TextInputType.number,
+                            decoration: const InputDecoration(
+                              border: UnderlineInputBorder(),
+                            ),
+                            inputFormatters: <TextInputFormatter>[
+                              FilteringTextInputFormatter.digitsOnly,
+                              ThousandsSeparatorInputFormatter(),
+                            ],
+                            onChanged: (value) {
+                              setState(() {
+                                try {
+                                  takerChannelAmount = int.parse(value.replaceAll(",", ""));
+                                } on Exception {
+                                  takerChannelAmount = 0;
+                                }
+                              });
+
+                              _formKey.currentState!.validate();
+                            },
+                            validator: (value) {
+                              if (value == null) {
+                                return "Enter channel amount";
                               }
-                            });
-                          },
-                        )
-                      ],
-                    )),
-                const SizedBox(height: 15),
-                ListTile(
-                    leading: addCircle("3"),
-                    title: const Text("Select your maker"),
-                    subtitle: const Text(
-                        "In this version you can only trade with 10101, but in the future you will be able to choose who to connect to.",
-                        style: TextStyle(color: Colors.grey))),
-                const SizedBox(height: 15),
-                ListTile(
-                    leading: addCircle("4"),
-                    title: const Text("Ready!"),
-                    subtitle: const Text(
-                        "Hit the 'Open Channel' button and the channel will be created in a few moments.",
-                        style: TextStyle(color: Colors.grey))),
-                const SizedBox(height: 15),
-              ],
+
+                              try {
+                                int intVal = int.parse(value.replaceAll(",", ""));
+
+                                if (intVal > maxTakerChannelAmount) {
+                                  return "Max channel amount of $maxTakerChannelAmount";
```

Comment:
Should we state `insufficient balance` instead?

Reply:
Not necessarily true, because we have the arbitrary constraint of 500,000 sats.